diff --git a/vamp-hostsdk/PluginHostAdapter.h b/vamp-hostsdk/PluginHostAdapter.h
index 2ca1d69..84b800f 100644
--- a/vamp-hostsdk/PluginHostAdapter.h
+++ b/vamp-hostsdk/PluginHostAdapter.h
@@ -106,6 +106,10 @@ public:
     FeatureSet process(const float *const *inputBuffers, RealTime timestamp);
 
     FeatureSet getRemainingFeatures();
+    
+    VampPluginHandle getPluginHandle() {
+        return m_handle;
+    }
 
 protected:
     void convertFeatures(VampFeatureList *, FeatureSet &);
diff --git a/vamp-hostsdk/PluginWrapper.h b/vamp-hostsdk/PluginWrapper.h
index 45e6880..ef52e01 100644
--- a/vamp-hostsdk/PluginWrapper.h
+++ b/vamp-hostsdk/PluginWrapper.h
@@ -123,6 +123,10 @@ public:
         if (pw) return pw->getWrapper<WrapperType>();
         return nullptr;
     }
+    
+    Plugin *getPlugin() {
+        return m_plugin;
+    }
 
     /**
      * Disown the wrapped plugin, so that we no longer delete it on
